src/main/java/BattleshipModel.java

```diff
@@ -81,61 +81,67 @@ public void aiFireHard()
         Coordinate left = null;
         Coordinate right = null;
 
-        if(shots.size() >=1) {
-            Coordinate previousHit = hits.get(hits.size() - 1);
+        if(shots.size() > 0) {
 
-            int x = previousHit.getAcross();
-            int y = previousHit.getDown();
+            if (hits.size() > 0) {
```

Comment:
Good catch! We forgot that this would fail if there were no hits in the first place since the function references a previous hit. If there isn't one, then it would error. Thanks!